velox/functions/prestosql/fuzzer/WindowFuzzerTest.cpp

```diff
@@ -0,0 +1,151 @@
+    "(e.g: --only \"min\" or --only \"sum,avg\").");
+
+namespace facebook::velox::exec::test {
+namespace {
+
+std::unordered_map<std::string, std::shared_ptr<InputGenerator>>
+getCustomInputGenerators() {
+  return {
+      {"min", std::make_shared<MinMaxInputGenerator>("min")},
+      {"min_by", std::make_shared<MinMaxInputGenerator>("min_by")},
+      {"max", std::make_shared<MinMaxInputGenerator>("max")},
+      {"max_by", std::make_shared<MinMaxInputGenerator>("max_by")},
+      {"approx_distinct", std::make_shared<ApproxDistinctInputGenerator>()},
+      {"approx_set", std::make_shared<ApproxDistinctInputGenerator>()},
+      {"approx_percentile", std::make_shared<ApproxPercentileInputGenerator>()},
+  };
+}
+
+} // namespace
+} // namespace facebook::velox::exec::test
+
+int main(int argc, char** argv) {
+  facebook::velox::aggregate::prestosql::registerAllAggregateFunctions(
+      "", false);
+  facebook::velox::window::prestosql::registerAllWindowFunctions();
+
+  ::testing::InitGoogleTest(&argc, argv);
+
+  // Calls common init functions in the necessary order, initializing
+  // singletons, installing proper signal handlers for better debugging
+  // experience, and initialize glog and gflags.
+  folly::Init init(&argc, &argv);
+
+  size_t initialSeed = FLAGS_seed == 0 ? std::time(nullptr) : FLAGS_seed;
+
+  auto duckQueryRunner =
+      std::make_unique<facebook::velox::exec::test::DuckQueryRunner>();
+  duckQueryRunner->disableAggregateFunctions({
+      "skewness",
+      // DuckDB results on constant inputs are incorrect. Should be NaN,
+      // but DuckDB returns some random value.
+      "kurtosis",
+      "entropy",
+  });
+
+  // List of functions that have known bugs that cause crashes or failures.
+  static const std::unordered_set<std::string> skipFunctions = {
+      // https://github.com/facebookincubator/velox/issues/3493
+      "stddev_pop",
+      // Lambda functions are not supported yet.
+      "reduce_agg",
+  };
+
+  // Functions whose results verification should be skipped. These can be
+  // functions that return complex-typed results containing floating-point
+  // fields.
+  // TODO: allow custom result verifiers.
+  static const std::unordered_map<
+      std::string,
+      std::shared_ptr<facebook::velox::exec::test::ResultVerifier>>
+      customVerificationFunctions = {
+          // Order-dependent functions.
+          {"approx_distinct", nullptr},
+          {"approx_set", nullptr},
+          {"approx_percentile", nullptr},
+          // Semantically inconsistent functions
+          {"skewness", nullptr},
+          {"kurtosis", nullptr},
+          {"entropy", nullptr},
+          // https://github.com/facebookincubator/velox/issues/6330
+          {"max_data_size_for_stats", nullptr},
+          {"sum_data_size_for_stats", nullptr},
+      };
+
+  static const std::unordered_set<std::string> orderDependentFunctions = {
+      // Window functions.
```

Comment:
I wonder if all window functions are order dependent. If so, perhaps, we don't need to hard code these.

Reply:
I think all the window functions in Presto are order-dependent. But when the fuzzer tests both window and aggregation functions, all function signatures are mixed together in a list, so we still need to add window function names to this list.

Reply:
The Fuzzer knows which functions are window functions and which are aggregate functions, right? Hence, it can just automatically "add" window functions to the list of order-dependent functions, no?